lib/matplotlib/_types.py

```diff
@@ -0,0 +1,165 @@
+"""
+Style desription information that is shared across unrelated classses.
+"""
+
+from enum import Enum
+from matplotlib import cbook
+
+
+def _deprecate_case_insensitive_join_cap(s):
+    s_low = s.lower()
+    if s != s_low:
+        if s_low in ['miter', 'round', 'bevel']:
+            cbook.warn_deprecated(
+                "3.3", message="Case-insensitive capstyles are deprecated "
+                "since %(since)s and support for them will be removed "
+                "%(removal)s; please pass them in lowercase.")
+        elif s_low in ['butt', 'round', 'projecting']:
+            cbook.warn_deprecated(
+                "3.3", message="Case-insensitive joinstyles are deprecated "
+                "since %(since)s and support for them will be removed "
+                "%(removal)s; please pass them in lowercase.")
+        # Else, error out at the check_in_list stage.
+    return s_low
+
+
+class JoinStyle(Enum):
+    """
+    Define how the connection between two line segments is drawn.
+
+    For a simple visual description of each *JoinStyle*, `view these docs
+    online <JoinStyle>`, or simply run `JoinStyle.demo`.
+
+    .. plot::
+        :alt: Demo of possible JoinStyle's
+
+        from matplotlib._types import JoinStyle
+        JoinStyle.demo()
+
+    For a more precise description, we first recall that lines in Matplotlib
+    are typically defined by a 1D `~.path.Path` and a finite ``linewidth``,
+    where the underlying 1D `~.path.Path` represents the center of the stroked
+    line.
+
+    By default, `~.backend_bases.GraphicsContextBase` defines the boundaries of
+    a stroked line to simply be every point within some radius,
+    ``linewidth/2``, away from any point of the center line. However, this
+    results in corners appearing "rounded", which may not be the desired
+    behavior if you are drawing, for example, a polygon or pointed star.
+
+    Matplotlib provides three options for defining how the corners where two
+    segments meet should be drawn. In short:
+
+        - *miter* is the "arrow-tip" style. Each boundary of the filled-in area
```

Comment:
You should not indent bullet points in ReST. That gives additional unwanted indent. Instead this should look like
~~~
    some text:

    - first item foo bar
      more text
    - second item
~~~

Yes, ReST is picky!

Reply:
Thanks for the heads up! I'm always impressed by how well everyone here keeps RST rules lined up. I switch so much between .md and .rst that I live in perpetual mix-up land.